web/src/compose_paste.ts

```diff
@@ -40,14 +40,74 @@ function image_to_zulip_markdown(
     return src ? "[" + title + "](" + src + ")" : (node.getAttribute("alt") ?? "");
 }
 
+// Returns 2 if there are two or more valid text nodes in the tree.
+function check_multiple_text_nodes(child_nodes: ChildNode[]): number {
+    let textful_nodes = 0;
+
+    for (const child of child_nodes) {
+        // We do not consider empty childNodes, comments and
+        // childNodes containing newlines.
+        if (child.nodeValue?.trim() === "" || child.nodeName === "#comment") {
+            continue;
+        }
+
+        if (child.nodeValue && child.nodeType === Node.TEXT_NODE) {
+            textful_nodes += 1;
+            if (textful_nodes >= 2) {
+                return textful_nodes;
+            }
+        }
+
+        textful_nodes += check_multiple_text_nodes([...child.childNodes]);
+
+        if (textful_nodes >= 2) {
+            return textful_nodes;
+        }
+    }
+
+    return textful_nodes;
+}
+
 function within_single_element(html_fragment: HTMLElement): boolean {
+    if (html_fragment.childNodes.length === 1) {
+        return true;
+    }
+    return false;
+}
```

Comment:
Why is the definition of this function changing? It doesn't look like you're still using it.